ray_julia_jll/gen/make.jl

```diff
@@ -0,0 +1,171 @@
+    end
+
+    m = match(GH_RELEASE_ASSET_PATH_REGEX, archive_uri.path)
+    if m === nothing
+        throw(ArgumentError(
+            "Artifact URL is not a GitHub release asset path: $(archive_uri)"
+        ))
+    end
+
+    # The `ghr` utility uses the local file name for the release asset. In order to have
+    # have the asset match the specified URL we'll temporarily rename the file.
+    org_archive_name = nothing
+    if basename(archive_path) != m[:file_name]
+        org_archive_name = basename(archive_path)
+        archive_path = mv(archive_path, joinpath(dirname(archive_path), m[:file_name]))
+    end
+
+    upload_to_github_release(m[:owner], m[:repo_name], commit, m[:tag], archive_path; kwargs...)
+
+    # Rename the archive back to the original name
+    if org_archive_name !== nothing
+        mv(archive_path, joinpath(dirname(archive_path), org_archive_name))
+    end
+
+    return nothing
+end
+
+
+function upload_to_github_release(owner, repo_name, commit, tag, path; token=ENV["GITHUB_TOKEN"])
+    # Based on: https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/d40ec617d131a1787851559ef1a9f04efce19f90/src/AutoBuild.jl#L487
+    # TODO: Passing in a directory path uploads multiple assets
+    # TODO: Would be nice to perform parallel uploads
+    cmd = ```
+        $(ghr()) \
+        -owner $owner \
+        -repository $repo_name \
+        -commitish $commit \
+        -token $token \
+        $tag $path
+    ```
+
+    run(cmd)
+end
+
+if abspath(PROGRAM_FILE) == @__FILE__
+
+    repo_path = abspath(joinpath(@__DIR__, "..", ".."))
+    pkg_url = remote_url(repo_path)
+
+    # Read Project.toml
+    jll_project_toml = joinpath(repo_path, "ray_julia_jll", "Project.toml")
+    jll_project = read_project(jll_project_toml)
+    jll_version = jll_project.version
+
+    # e.g. ray_julia.v0.1.0.aarch64-apple-darwin-libgfortran5-cxx11-julia_version+1.9.2
+    host_platform = BinaryPlatforms.host_triplet()
+    tarball_name = "ray_julia.v$jll_version.$host_platform.tar.gz"
+
+    # Build JLL
+    # TODO: execute inside a python venv
+    Pkg.build("ray_julia_jll"; verbose=true)
+    compiled_dir = joinpath(repo_path, "ray_julia_jll", "deps", "bazel-bin")
+
+    @info "Creating tarball $tarball_name"
+    tarball_path = joinpath(tempdir(), tarball_name)
+
+    create_tarball(readlink(compiled_dir), tarball_path)
+
+    tag = "v$(jll_version)"
+    artifact_url = "$(pkg_url)/releases/download/$(tag)/$(basename(tarball_path))"
+
+    artifacts_toml = joinpath(repo_path, "Artifacts.toml")
+    bind_artifact!(
+        artifacts_toml,
+        "ray_julia_jll",
+        tree_hash_sha1(tarball_path);
+        download_info=[(artifact_url, sha256sum(tarball_path))],
+        force=true
+    )
+
+    # TODO: Ensure no other files are staged before committing
+    @info "Committing and pushing changes to Artifacts.toml"
+    branch = "gm/test"  # TODO - pass as function arg
```

Comment:
Since this won't be pusing to`main` the branch name will probably have to be passed in as an arg to the script.

Moreover the branch will have to be repeatedly `commited` to if we want to bundle each platform/julia version into the same PR

Reply:
I've opted to push the Artifacts.toml changes to the currently checked out branch. 
A few things we should check in this regard:

- the branch is clean (no unstaged commits)
- the Artifacts.toml is the only change staged/committed
- probably something about it being a new branch...